sql/src/main/java/io/crate/protocols/postgres/PostgresWireProtocol.java

```diff
@@ -237,24 +235,12 @@ private ReadyForQueryCallback(Channel channel) {
         }
 
         @Override
-        public void accept(Object o, Throwable t) {
-            if (t == null) {
-                onSuccess(o);
-            } else {
-                onFailure(t);
-            }
-        }
-
-        private void onSuccess(@Nullable Object result) {
+        public void accept(Object result, Throwable t) {
             if (result == null || result.equals(Boolean.FALSE)) {
                 // only send ReadyForQuery if query was not interrupted
                 Messages.sendReadyForQuery(channel);
             }
         }
```

Comment:
> `onFailure` also called `sendReadyForQuery`, so the `t != null` check would be wrong. 

I don't understand.`onFailure` implies `t != null`. As far as I see the code I posted preserves the exact semantics.

If we assume `t XOR result` (you're right, should be guaranteed), then we can indeed simplify further like you did.